akka-remote/src/main/scala/akka/remote/artery/ArteryTcpTransport.scala

```diff
@@ -0,0 +1,365 @@
+      inboundConnectionsKillSwitch = KillSwitches.shared("inboundConnectionsKillSwitch")
+      unbind().onComplete { _ ⇒
+        runInboundStreams()
+      }
+    }
+    attachInboundStreamRestart("Inbound streams", completed, restart)
+  }
+
+  private def runInboundControlStream(): (Sink[EnvelopeBuffer, NotUsed], Future[Done]) = {
+    if (isShutdown) throw ArteryTransport.ShuttingDown
+
+    val (hub, ctrl, completed) =
+      MergeHub.source[EnvelopeBuffer].addAttributes(Attributes.logLevels(onFailure = LogLevels.Off))
+        .via(inboundFlow(settings, NoInboundCompressions))
+        .toMat(inboundControlSink)({ case (a, (c, d)) ⇒ (a, c, d) })
+        .run()(controlMaterializer)
+    attachControlMessageObserver(ctrl)
+    implicit val ec = materializer.executionContext
+    updateStreamMatValues(controlStreamId, completed)
+
+    (hub, completed)
+  }
+
+  private def runInboundOrdinaryMessagesStream(): (Sink[EnvelopeBuffer, NotUsed], Future[Done]) = {
+    if (isShutdown) throw ArteryTransport.ShuttingDown
+
+    val (inboundHub: Sink[EnvelopeBuffer, NotUsed], inboundCompressionAccess, completed) =
+      if (inboundLanes == 1) {
+        MergeHub.source[EnvelopeBuffer].addAttributes(Attributes.logLevels(onFailure = LogLevels.Off))
+          .viaMat(inboundFlow(settings, _inboundCompressions))(Keep.both)
+          .toMat(inboundSink(envelopeBufferPool))({ case ((a, b), c) ⇒ (a, b, c) })
+          .run()(materializer)
+
+      } else {
+        // TODO perhaps a few more things can be extracted and DRY with AeronUpdTransport.runInboundOrdinaryMessagesStream
+        val hubKillSwitch = KillSwitches.shared("hubKillSwitch")
+
+        val source: Source[InboundEnvelope, (Sink[EnvelopeBuffer, NotUsed], InboundCompressionAccess)] =
+          MergeHub.source[EnvelopeBuffer].addAttributes(Attributes.logLevels(onFailure = LogLevels.Off))
+            .via(hubKillSwitch.flow)
+            .viaMat(inboundFlow(settings, _inboundCompressions))(Keep.both)
+            .via(Flow.fromGraph(new DuplicateHandshakeReq(inboundLanes, this, system, envelopeBufferPool)))
+
+        val (inboundHub, compressionAccess, hub) =
+          source
+            .toMat(Sink.fromGraph(new FixedSizePartitionHub[InboundEnvelope](inboundLanePartitioner, inboundLanes,
+              settings.Advanced.InboundHubBufferSize)))({ case ((a, b), c) ⇒ (a, b, c) })
+            .run()(materializer)
+
+        val lane = inboundSink(envelopeBufferPool)
+        val completedValues: Vector[Future[Done]] =
+          (0 until inboundLanes).map { _ ⇒
+            hub.toMat(lane)(Keep.right).run()(materializer)
+          }(collection.breakOut)
+
+        import system.dispatcher
+
+        // tear down the upstream hub part if downstream lane fails
+        // lanes are not completed with success by themselves so we don't have to care about onSuccess
+        Future.firstCompletedOf(completedValues).failed.foreach { reason ⇒ hubKillSwitch.abort(reason) }
```

Comment:
is this logged else where when this happens?

Reply:
it's logged in `attachInboundStreamRestart`